jbi/models.py

```diff
@@ -391,3 +392,41 @@ class BugzillaApiResponse(BaseModel):
 
     faults: Optional[list]
     bugs: Optional[list[BugzillaBug]]
+
+
+class JiraContext(BaseModel):
+    """Logging context about Jira"""
+
+    project: str
+    issue: Optional[str]
+
+
+BugId = TypedDict("BugId", {"id": Optional[int]})
+
+
+class LogContext(BaseModel):
+    """Generic log context throughout JBI"""
+
+    def set(self, **kwargs):
+        """Set multiple attributes at once"""
+        for k, value in kwargs.items():
+            setattr(self, k, value)
+        return self
```

Comment:
I'm seeing that another pattern for updating a model is to use a [model's `copy` method](https://pydantic-docs.helpmanual.io/usage/exporting_models/#modelcopy) with an `update` kwarg:
```python
log_context = log_context.copy(update={action: action})
```
I like that this is a more "models are immutable" approach, and it also re-validates the data.

If you want to stick with the approach you have here, could we:
- maybe rename the method `update`? Visually, `set` with a bunch of kwargs says "initialization", but maybe that's just me.
- validate the model after we update the data?

Reply:
Oh! I promise I looked for something like this !